site/en/blog/crx-scripting-api/index.md

```diff
@@ -0,0 +1,209 @@
+functionality where we're thinking about scripting in extensions more broadly.
+
+On the executeScript front, we also wanted to expand what this API could do beyond what the Tabs API version supported. More specifically, we wanted to support functions and arguments, more easily target specific frames, and target non-"tab" contexts.
+
+## Changes in Manifest V3
+
+In the previous section I mentioned that we're planning to add dynamic content script support. While
+I'd love to dig more into what this might look like or how it might work, design goals, or
+capabilities, unfortunately we're not quite ready to share those details.
+
+In the remainder of this post, I'd like to take a closer look at the similarities and differences between `chrome.tabs.executeScript` and `chrome.scripting.executeScript`.
+
+While considering how the platform would need to evolve in light of remotely hosted code
+restrictions, we wanted to find a balance between the raw power of arbitrary code execution and only
+allowing static content scripts. The solution we struck on was to allow extensions to inject a
+function as a content script and to pass an array of values as arguments.
+
+Let's take a quick look at an (oversimplified) example. Say we wanted to inject a script that
+greeted the user by name when the user clicks the extension's action button (icon in the toolbar).
+In Manifest V2, we could dynamically construct a code string and execute that script in the current
+page.
+
+```js
+// Manifest V2 extension
+let givenName = "Default";
+chrome.browserAction.onClicked.addListener((tab) => {
+  chrome.tabs.executeScript({
+    code: `alert(\`Hello, ${givenName}\`)`
+  });
+});
+```
+
+While Manifest V3 extensions can't use code that isn't bundled with the extension, our goal was to
+preserve some of the dynamism that arbitrary code blocks enabled for Manifest V2 extensions. The
+function and arguments approach makes it possible for Chrome Web Store reviewers, users, and other
+interested parties to more accurately assess the risks an extension poses while also allowing
+developers to modify an extension's runtime behavior based on user settings or application state.
+
+```js
+// Manifest V3 extension
+let givenName = "Default";
+
+function greetUser(name) {
+  alert(`Hello, ${name}`);
+}
+chrome.action.onClicked.addListener((tab) => {
+  chrome.tabs.executeScript({
+    target: { tabId: tab.id },
+    func: greetUser,
+    args: [givenName],
+  });
+});
+```
+
+We also wanted to make frames more of a first-class citizen of the API. The Manifest V2 version of
+`executeScript` allowed developers to target either all frames in a tab or a specific frame in the
+tab. In Manifest V3, we replaced the optional `frame` string property in the options object with an
+optional `frames` array of strings.
+
+
+```js
+// Manifest V2 extension
+chrome.action.onClicked.addListener((tab) => {
+  chrome.scripting.executeScript({
```

Comment:
This MV2 extension is using scripting.executeScript() and action, but should be using tabs.executeScript and browserAction.  (This will also entail updating the arguments passed in the executeScript() call)

Reply:
Fixed.